app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
The main purpose of `search_redirect` is to clean up the url before going to `index`. Of course, I can have a bunch of checks and conditional redirects in `index`, but it would be much easier and faster to edit the parameters in a separate action before moving on to `index`. (That is, unless you want a url like `https://www.redstoner.com/forums/threads?id=Search+All+Threads&label=Label&title=This` instead of `https://www.redstoner.com/forums/threads?title=This`)